packages/web-ui/components/admin/update-roles-form.tsx

```diff
@@ -0,0 +1,172 @@
+'use client';
+
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import Endpoints from '@/api/endpoints';
+import { superAdminApiClient } from '@/api/superAdminApiClient';
+import { SuperAdminUpdateUserRoleRequestSchema } from '@/contract/auth/super-admin-update-role.request.dto';
+import { SuperAdminUpdateUserRoleRequestDto } from '@/contract/auth/super-admin-update-role.request.dto.d';
+import { getAppRoles } from '@/utils/get-app-roles';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { getCookie } from 'cookies-next';
+import { Settings } from 'lucide-react';
+import { useForm } from 'react-hook-form';
+
+import { Button } from '@/components/ui/button';
+import { Checkbox } from '@/components/ui/checkbox';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import { Separator } from '@/components/ui/separator';
+import { useToast } from '@/components/ui/use-toast';
+
+interface UpdateRolesFormProps {
+  userId: string;
+  defaultRoles: string[];
+}
+
+const roles = getAppRoles().map((role) => ({ id: role, label: role }));
+
+export function UpdateRolesForm({
+  userId,
+  defaultRoles,
+}: UpdateRolesFormProps) {
+  const [open, setOpen] = useState(false);
+  const { toast } = useToast();
+  const router = useRouter();
+  const adminCredentialsCookie = getCookie('__admin');
+  let adminCredentials: { email: string; password: string };
+  if (adminCredentialsCookie) {
+    adminCredentials = JSON.parse(adminCredentialsCookie!.toString());
```

Comment:
We don't need the `!`, we are already checking its existence

Reply:
well spotted 👀